scripts/simplify_network.py

```diff
@@ -161,12 +161,13 @@ def _prepare_connection_costs_per_link(n, costs, config):
 
     connection_costs_per_link = {}
 
-    if not n.links.loc[n.links.carrier == "DC"].empty:
-        dc_lengths = n.links.length
-        unterwater_fractions = n.links.underwater_fraction
-    elif not n.lines.loc[n.lines.carrier == "DC"].empty:
+    # initialize dc_lengths and underwater_fractions by the hvdc_as_lines option
+    if config["electricity"]["hvdc_as_lines"]:
         dc_lengths = n.lines.length
         unterwater_fractions = n.lines.underwater_fraction
+    else:
+        dc_lengths = n.links.length
+        unterwater_fractions = n.links.underwater_fraction
```

Comment:
Great fix! I think, it looks cleaner now :)

As a side comment, `dc_length` naming looks quite misleading here... Would you mind to remove "dc_" prefix from it?

Reply:
Actually, the reason of these lines it to capture the lines that represent DC lines, be them represented as lines or links.
I think the naming should be ok.
Do you think it needs revision?

Reply:
> Actually, the reason of these lines it to capture the lines that represent DC lines, be them represented as lines or links.
> 
> I think the naming should be ok.
> 
> Do you think it needs revision?

I needed some time to remember what does dc_lines mean, when looking into the code :) But naming is generally tricky, and if you feel it's alright then it most likely is :)